lib/src/annotation.dart

```diff
@@ -150,7 +150,17 @@ dynamic _createFromConstructor(
       // field assigned in the object. Then we can take the field value and
       // set it as the argument value
 
+      // If the annotation class's constructor has no initializers and just
+      // forwards to a `super` class constructor, then there is no need
+      // to add a ctor argument name as a field value.
+      //
+      // In this case `singleWhere` would throw a StateError.
+      if (ctor.constantInitializers.length == 1 && ctor.constantInitializers.first is SuperConstructorInvocation)
+        continue;
```

Comment:
I don't quite understand under which scenario this happens, in the test provided there are zero parameters so we wouldn't reach this code. If you could add a test outlining the case that this is handling that would help clarify things :).

It also doesn't seem like skipping a parameter entirely is ever the right thing to do?